### PR TITLE
Catch ZstdError in jlap/repodata.json.zst handling

### LINKED ISSUE #13558: Fallback to repodata.json if repodata.json.zst raises a zstandard exception
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

conda should be able to fall back to repodata.json if repodata.json.zst is not zst, not only if it 404's.

### Conda Info

24.1.0

### PR DESCRIPTION
**Pull request by dholth**
### Description

Fix #13558

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: conda/gateways/repodata/jlap/fetch.py [hunk 1]
```diff
@@ -304,7 +304,10 @@ def request_url_jlap_state(
                     )
                 else:
                     raise JlapSkipZst()
-            except (JlapSkipZst, HTTPError) as e:
+            except (JlapSkipZst, HTTPError, zstandard.ZstdError) as e:
+                if isinstance(e, zstandard.ZstdError):
+                    # will this include the token
```
Comment (jezdez) [inline conda/gateways/repodata/jlap/fetch.py:309]: I think that's reasonable, so maybe sanitize it? Or leave out the URL?

**Review by jezdez — APPROVED** [review]
Not sure if we actually need the debugging statement, but I concur it's good to catch zstandard decompression exceptions

**Comment by dholth** [comment]
In general this entire code path could be more clear about whether the user's web server is configured correctly, and which variant they wind up getting. If we don't say something then the user would not know that they have a "404 page that returns a 200 code" for `repodata.json.zst`

**Comment by jezdez** [comment]
Then lets make this and the other possible cases result in more specific (and actionable) logging statement? Info level?

**Comment by dholth** [comment]
Improved log message with sanitized URL, and a bonus sanitized URL in a similar log message.

File: conda/gateways/repodata/jlap/fetch.py (hunk 1 above)
Comment (dholth) [reply conda/gateways/repodata/jlap/fetch.py]: Improved.

File: tests/core/test_solve.py [hunk 2]
```diff
@@ -1895,7 +1895,7 @@ def test_python2_update(tmpdir):
         )
         assert convert_to_dist_str(final_state_1) == order1
 
-    specs_to_add = (MatchSpec("python=3"),)
+    specs_to_add = (MatchSpec("python=3.7"),)
```
Comment (dholth) [inline tests/core/test_solve.py:1898]: This part started failing for no runners seemingly on its own.

Comment (dholth) [reply tests/core/test_solve.py:1898]: https://github.com/conda/conda-libmamba-solver/issues/317

File: tests/core/test_solve.py [hunk 3]
```diff
@@ -1949,6 +1949,11 @@ def test_python2_update(tmpdir):
                     "channel-4::pycosat-0.6.3-py37h14c3975_0",
                 )
             )
+
+            pprint("Important parts")
+            pprint(set(important_parts))
+            pprint("Full solution")
+            pprint(set(full_solution))
```
Comment (jezdez) [inline tests/core/test_solve.py:1956]: Do we need those?

**Pull request merged**